aten/src/ATen/native/MetaTensor.cpp

```diff
@@ -7,34 +7,29 @@ namespace native {
 // Will be promoted to a public API later, but not now
 Tensor empty_meta(
   IntArrayRef size,
-  const TensorOptions& options_,
-  c10::optional<c10::MemoryFormat> optional_memory_format
+  c10::optional<ScalarType> dtype,
+  c10::optional<Layout> layout,
+  c10::optional<Device> device,
+  c10::optional<bool> pin_memory,
+  c10::optional<c10::MemoryFormat> memory_format
```

Comment:
It's cool that we managed to kill the duping, though we had better have handled it somewhere else now :)

Reply:
you mean the assertion? Hm I guess that would be the task of the C++ frontend now since that's where we unpack the TensorOptions. I don't think it's currently doing it though.

Reply:
ah good point, the C++ frontend actually does do it already.